fineract-provider/src/main/java/org/apache/fineract/infrastructure/creditbureau/service/ThitsaWorksCreditBureauIntegrationWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,420 @@
+                "SubscriptionId");
+        CreditBureauConfiguration subscriptionKeyData = this.configDataRepository.getCreditBureauConfigData(creditBureauId,
+                "SubscriptionKey");
+        CreditBureauConfiguration userNameData = this.configDataRepository.getCreditBureauConfigData(creditBureauId, "Username");
+        CreditBureauConfiguration passwordData = this.configDataRepository.getCreditBureauConfigData(creditBureauId, "Password");
+
+        String subscriptionId = "";
+        String subscriptionKey = "";
+        String userName = "";
+        String password = "";
+
+        try {
+            subscriptionId = subscriptionIdData.getValue();
+            subscriptionKey = subscriptionKeyData.getValue();
+            userName = userNameData.getValue();
+            password = passwordData.getValue();
+        } catch (NullPointerException ex) {
+            throw new PlatformDataIntegrityException("Credit Bureau Configuration is not available",
+                    "Credit Bureau Configuration is not available" + ex);
+        }
+
+        String token = null;
+        if (!"".equals(subscriptionId) && !"".equals(subscriptionKey) && !"".equals(userName) && !"".equals(password)) {
+            token = createToken(userName, password, subscriptionId, subscriptionKey, creditBureauId);
+        } else {
+            throw new PlatformDataIntegrityException("Credit Bureau Configuration is not available",
+                    "Credit Bureau Configuration is not available");
+        }
+
+        // will use only "NRC" part of code from common http method to get data based on nrc
+        String process = "NRC";
+        CreditBureauConfiguration SearchURL = this.configDataRepository.getCreditBureauConfigData(creditBureauId, "searchurl");
+        String url = SearchURL.getValue();
+        String result = this.httpConnectionMethod(process, nrcId, userName, password, subscriptionKey, subscriptionId, url, token, 0L,
+                null);
+
+        // after fetching the data from httpconnection it will be come back here for fetching UniqueID from data
+        if (process.equals("NRC")) {
+
+            JsonObject reportObject = JsonParser.parseString(result).getAsJsonObject();
+
+            JsonElement element = reportObject.get("Data");
+
+            if (element.isJsonNull()) {
+                String ResponseMessage = reportObject.get("ResponseMessage").getAsString();
+                handleAPIIntegrityIssues(ResponseMessage);
+            }
+
+            // to fetch the Unique ID from Result
+            JsonObject jsonObject = JsonParser.parseString(result).getAsJsonObject();
+            Long uniqueID = 0L;
+            try {
+                JsonArray jArray = jsonObject.getAsJsonArray("Data");
+                JsonObject jobject = jArray.get(0).getAsJsonObject();
+                String uniqueIdString = jobject.get("UniqueID").toString();
```

Comment:
extract this logic in method. Can we validate uniqueId, instead of depending on IndexOutOfBoundsExcp